src/main/java/org/hmhb/authentication/DefaultJwtAuthenticationService.java

```diff
@@ -0,0 +1,157 @@
+package org.hmhb.authentication;
+
+import javax.annotation.Nonnull;
+import javax.crypto.spec.SecretKeySpec;
+import javax.servlet.http.HttpServletRequest;
+import javax.xml.bind.DatatypeConverter;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.hmhb.user.HmhbUser;
+import org.hmhb.user.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class DefaultJwtAuthenticationService implements JwtAuthenticationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAuthenticationService.class);
+
+    private final Environment environment;
+    private final UserService userService;
+
+    @Autowired
+    public DefaultJwtAuthenticationService(
+            @Nonnull Environment environment,
+            @Nonnull UserService userService
+    ) {
+        this.environment = requireNonNull(environment, "environment cannot be null");
+        this.userService = requireNonNull(userService, "userService cannot be null");
+    }
+
+    private String getDomain() {
+        return environment.getProperty("hmhb.jwt.domain");
+    }
+
+    private String getSecret() {
+        return environment.getProperty("hmhb.jwt.secret");
+    }
+
+    private Date getTokenExpirationDate() {
+        /* one week */
+        return new Date(System.currentTimeMillis() + 7 * 24 * 60 * 60 * 1000);
+    }
+
+    @Override
+    public String generateJwtToken(
+            @Nonnull HmhbUser user
+    ) {
+        Map<String, Object> claims = new HashMap<>();
+        claims.put("admin", user.isAdmin());
+        claims.put("email", user.getEmail());
+
+        String domainId = getDomain();
+        String secret = getSecret();
+
+        String token = Jwts.builder()
+                .setClaims(claims)
```

Comment:
TODO: add a comment explaining that you have to call setClaims first or the builder throws away your previous .sets.